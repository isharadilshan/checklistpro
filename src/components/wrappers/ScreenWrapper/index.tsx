import React from 'react';
import { View } from 'react-native';
import createStyle from './style';

interface ScreenWrapperProps {
  children: React.ReactNode;
  noPaddings?: boolean;
}

const ScreenWrapper: React.FC<ScreenWrapperProps> = ({
  children,
  noPaddings,
}) => {
  const styles = createStyle();
  return (
    <View
      style={[styles.wrapperStyle, { paddingHorizontal: noPaddings ? 0 : 16 }]}
    >
      {children}
    </View>
  );
};

export default ScreenWrapper;
