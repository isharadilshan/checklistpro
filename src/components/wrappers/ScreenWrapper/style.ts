import { ViewStyle, StyleSheet } from 'react-native';

interface Style {
  wrapperStyle: ViewStyle;
}

export default () => {
  return StyleSheet.create<Style>({
    wrapperStyle: {
      flex: 1,
      backgroundColor: '#141E30',
    },
  });
};
