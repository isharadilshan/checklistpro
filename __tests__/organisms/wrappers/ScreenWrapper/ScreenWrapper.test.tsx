import React from 'react';
import {describe, expect} from '@jest/globals';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import ScreenWrapper from '../../../../src/components/wrappers/ScreenWrapper';

configure({adapter: new Adapter()});

describe('when rendering', () => {
  const wrapper = shallow(<ScreenWrapper />);

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
