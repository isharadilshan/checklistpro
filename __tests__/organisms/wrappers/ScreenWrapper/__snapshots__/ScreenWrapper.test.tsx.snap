// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering should render a component matching the snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#141E30",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
/>
`;
